var/spack/repos/builtin/packages/aomp/package.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+tools_url = 'https://github.com/ROCm-Developer-Tools'
+compute_url = 'https://github.com/RadeonOpenCompute'
+aomp_sha = 'e4526489833896bbc47ba865e0d115fab278ce269789a8c99a97f444595f5f6a'
+devlib_sha = 'dce3a4ba672c4a2da4c2260ee4dc96ff6dd51877f5e7e1993cb107372a35a378'
+llvm_sha = 'b4fd7305dc57887eec17cce77bbf42215db46a4a3d14d8e517ab92f4e200b29d'
+flang_sha = 'cc27f8bfb49257b7a4f0b03f4ba5e06a28dcb6c337065c4201b6075dd2d5bc48'
+extras_sha = '5dbf27f58b8114318208b97ba99a90483b78eebbcad4117cac6881441977e855'
+hip_sha = '86eb7749ff6f6c5f6851cd6c528504d42f9286967324a50dd0dd54a6a74cacc7'
+vdi_sha = 'b21866c7c23dc536356db139b88b6beb3c97f58658836974a7fc167feb31ad7f'
+opencl_sha = '8963fcd5a167583b3db8b94363778d4df4593bfce8141e1d3c32a59fb64a0cf6'
+
+
+class Aomp(Package):
+    """llvm openmp compiler from AMD."""
+
+    homepage = tools_url + "/aomp"
+    url      = tools_url + "/aomp/archive/rocm-3.5.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala', 'estewart08']
+    version('3.5.0', sha256=aomp_sha)
+
+    depends_on('cmake@3.5.2', type='build')
+    depends_on('binutils', type='build')
+    depends_on('rocm-device-libs@3.5:', type='build', when='@3.5:')
+    depends_on('hsakmt-roct@3.5:', type='build', when='@3.5:')
+    depends_on('hsa-rocr-dev@3.5:', type='build', when='@3.5:')
+    depends_on('comgr@3.5:', type='build', when='@3.5:')
+    depends_on('mesa~llvm@18.3:', type=('build', 'link'), when='@3.5:')
+    depends_on('py-setuptools@44.1.0', when='@3.5:')
+    depends_on('python@2.7.18', when='@3.5:')
+    depends_on('perl-data-dumper', type='build', when='@3.5:')
+    depends_on('gawk', type='build', when='@3.5:')
```

Comment:
```suggestion
    depends_on('awk', type='build', when='@3.5:')
```
This allows you to use the virtual dependency.

Also, most of these don't need `when='@3.5:'` since you only have a single version in the package and it's always covered by that.